src/Function.cpp

```diff
@@ -574,6 +605,9 @@ Function::OutputForwardDecl(std::ostream &out)
 		int power = rnd_upto(16);
 		out << " __attribute__((aligned(" << (1 << power) << ")))";
 	}
+	if(func_attr_flag){
+		for_each(attributes.begin(), attributes.end(), std::bind2nd(std::ptr_fun(output_attr), out));
```

Comment:
Is this the correct way to call `output_attr` using respective object from `attributes` and pass `out` as parameter.

Reply:
Please use normal `for` loop for now.

Reply:
Here, during `OutputForwardDecl`, the derived objects are destroyed. It causes `segmentation fault` due to acessing `output_attr` using such objects.

Approach to solve above - 
During `make_random`, attributes generated using `generate_func_attr` can be stored with `vector` associated with function. In output phase, this vector can be print.